### DISCUSSION TITLE
Option to enable "Wrap lines" in log details only

### DISCUSSION THREAD

**Discussion by Mario-Hofstaetter** [grafana/grafana#45409] [Observability]
**What would you like to be added**:

It would be great to have the possibility to "Wrap lines" only selected fields (or labels) in the "Log details" view.
So a user can line-wrap all fields (or selected fields), without changing the view of the whole log list.

Maybe this can be added in a similar fashion as the "ad hoc statistics", filter and "view" buttons:

![grafik](https://user-images.githubusercontent.com/33002073/114286352-5777bb80-9a5e-11eb-86b1-d6b1d958db2d.png)

**Why is this needed**:

When dealing with very long log lines (legacy apps etc.), usually "Wrap lines" is disabled for much better handling / overview.
(This is just two log lines..)
![grafik](https://user-images.githubusercontent.com/33002073/114286397-9c035700-9a5e-11eb-9c24-808253eb9109.png)

When a log messages is expanded, it is also not quite possible to use the horizontal scrollbar, **because it is at the bottom of the explore page**. This makes getting to the "end" of the message field suprisingly difficult.

![FPgv6PW821](https://user-images.githubusercontent.com/33002073/114286522-cb669380-9a5f-11eb-87be-c05093683fa3.gif)

(As a workaround, Keyboard Arrow Right can be used to scroll horizontally)

Because of this users often need to add additional filters and/or decrease "line limit" so the horizontal scrollbar is accessible on screen.

**Comment by hugohaggmark** [comment]
Adding this to @grafana/observability-squad backlog.

**Comment by hterik** [comment]
The first use case described would also be improved drastically if detected fields could be rendered in tabular form, as described in https://github.com/grafana/grafana/issues/31481

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
